src/connect.rs

```diff
@@ -403,6 +404,26 @@ impl Connector {
                         is_proxy: false,
                         tls_info: false,
                     });
+                } else if proxy_dst.scheme() == Some(&Scheme::HTTP) {
+                    let host = dst.host().to_owned();
+                    let port = dst.port().map(|p| p.as_u16()).unwrap_or(443);
+                    let mut http = http.clone();
+                    let conn = http.call(proxy_dst).await?;
+                    log::trace!("tunneling HTTPS over HTTP proxy");
+                    let tunneled = tunnel(
```

Comment:
Would someone be able to help me? I'm not sure if this logic is correct.

It connects to the proxy server using `CONNECT`, so it seems bit different from HTTP proxy